src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/EmbeddingService.java

```diff
@@ -9,5 +9,6 @@
  * of chosen model.
  */
 public interface EmbeddingService {
-  Uni<List<float[]>> vectorize(List<String> texts, Optional<String> apiKeyOverride);
+  Uni<List<float[]>> vectorize(
```

Comment:
Needs Javadoc to explain "isWrite" at least?

Reply:
Actually, I think name "isWrite" is probably not very descriptive, not sure what'd be better: I think it means "used for vectorizing document to insert"? (vs "read" meaning "used for queries"?).

Is there likelihood we might need more than 2 types in future?

Reply:
Make sense, changed it to a enum with values `INDEX` vs `SEARCH`